scripts/effectManager/models.js

```diff
@@ -0,0 +1,93 @@
+import { CUSTOM_EFFECT_MODE_LID, CUSTOM_EFFECT_MODE_NAME } from "../consts.js";
+
+/* -------------------------------------------- */
+/*  Schema                                      */
+/* -------------------------------------------- */
+
+export const schemaFolder = new foundry.data.fields.SchemaField({
+    name: new foundry.data.fields.StringField({
+        label: "Name",
+    }),
+
+    isCollapsed: new foundry.data.fields.BooleanField({
+        initial: false,
+    }),
+});
+
+export const schemaCustomEffect = new foundry.data.fields.SchemaField({
+    // TODO(v12) switch to `foundry.data.fields.DocumentUUIDField`
+    macroUuid: new foundry.data.fields.StringField({
+        type: "Macro",
+        label: "lancer-weapon-fx.effectManager.fields.macroUuid.label",
+        nullable: true,
+    }),
+
+    // TODO(v12) switch to `foundry.data.fields.DocumentUUIDField`
+    folderId: new foundry.data.fields.StringField({ nullable: true }),
+
+    mode: new foundry.data.fields.NumberField({
+        integer: true,
+        choices: [CUSTOM_EFFECT_MODE_NAME, CUSTOM_EFFECT_MODE_LID],
+        nullable: true,
+        initial: CUSTOM_EFFECT_MODE_NAME,
+    }),
+
+    // region Fields specific to "name" mode
+    itemName: new foundry.data.fields.StringField({
+        label: "Weapon Name",
+        nullable: true,
+    }),
+    // endregion
+
+    // region Fields specific to "LID" mode
+    itemLid: new foundry.data.fields.StringField({
+        label: "Lancer ID",
+        nullable: true,
+    }),
+    // endregion
+});
+
+/* -------------------------------------------- */
+/*  Data Model                                  */
+/* -------------------------------------------- */
+
+/**
+ * Notes:
```

Comment:
This is not ideal, but it works. In v12 the implementation does not change, but they are committed to expanding datamodel/field support so maybe in v13+ this will need a revisit!